src/client/imagegen.h

```diff
@@ -0,0 +1,78 @@
+/*
+Minetest
+Copyright (C) 2010-2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#pragma once
+
+#include <IImage.h>
+#include <string>
+#include "settings.h"
+
+// This file is only for internal generation/modification of images.
+// Use texturesource.h instead to handle textures.
+
+// A cache used for storing source images.
+class SourceImageCache {
+public:
+	~SourceImageCache();
+
+	void insert(const std::string &name, video::IImage *img, bool prefer_local);
+
+	video::IImage* get(const std::string &name);
+
+	// Primarily fetches from cache, secondarily tries to read from filesystem
+	video::IImage *getOrLoad(const std::string &name);
+private:
+	std::map<std::string, video::IImage*> m_images;
+};
+
+/*
+ * Generates and caches images.
+ * The image name defines the image by filename and texture modifiers.
+*/
+struct ImageGen {
```

Comment:
generator sounds kinda wrong
how about `ImageSource` to mirror the other class?

Reply:
Renamed it to `ImageSource`.